src/test/java/com/iexec/standalone/metric/MetricsServiceTests.java

```diff
@@ -0,0 +1,292 @@
+
+        when(preComputeDurationsService.getAggregatedDurations()).thenReturn(new AggregatedDurations(
+                2,
+                1_000.0,
+                1_500.0,
+                1_250.0
+        ));
+        when(appComputeDurationsService.getAggregatedDurations()).thenReturn(new AggregatedDurations(
+                2,
+                3_000.0,
+                3_500.0,
+                3_250.0
+        ));
+        when(postComputeDurationsService.getAggregatedDurations()).thenReturn(new AggregatedDurations(
+                2,
+                2_000.0,
+                2_500.0,
+                2_250.0
+        ));
+
+        final WorkerMetrics workerMetrics = metricsService.getWorkerMetrics();
+
+        final WorkerMetrics expectedWorkerMetrics = new WorkerMetrics(
+                new AggregatedDurations(2, 1_000.0, 1_500.0, 1_250.0),
+                new AggregatedDurations(2, 3_000.0, 3_500.0, 3_250.0),
+                new AggregatedDurations(2, 2_000.0, 2_500.0, 2_250.0),
+                new AggregatedDurations(2, 6_000.0, 7_500.0, 6_750.0)
+        );
+
+        System.out.println(workerMetrics);
+        System.out.println("-----");
+        System.out.println(expectedWorkerMetrics);
+        assertThat(workerMetrics).isEqualTo(expectedWorkerMetrics);
+    }
+    // endregion
+
+    // region getCompleteComputeMetrics
+    @Test
+    void shouldGetCompleteComputeMetrics() {
+        when(appComputeDurationsService.getChainTaskIds()).thenReturn(List.of(CHAIN_TASK_ID_1, CHAIN_TASK_ID_2));
+
+        // First task has completed
+        when(preComputeDurationsService.getDurationForTask(CHAIN_TASK_ID_1)).thenReturn(Optional.of(1_000L));
+        when(appComputeDurationsService.getDurationForTask(CHAIN_TASK_ID_1)).thenReturn(Optional.of(3_000L));
+        when(postComputeDurationsService.getDurationForTask(CHAIN_TASK_ID_1)).thenReturn(Optional.of(2_000L));
+
+        // Second task has completed
+        when(preComputeDurationsService.getDurationForTask(CHAIN_TASK_ID_2)).thenReturn(Optional.of(1_500L));
+        when(appComputeDurationsService.getDurationForTask(CHAIN_TASK_ID_2)).thenReturn(Optional.of(3_500L));
+        when(postComputeDurationsService.getDurationForTask(CHAIN_TASK_ID_2)).thenReturn(Optional.of(2_500L));
+
+        final AggregatedDurations completeComputeMetrics = metricsService.getCompleteComputeMetrics();
+
+        assertThat(completeComputeMetrics.getMinDuration()).isEqualTo(6_000.0);
+        assertThat(completeComputeMetrics.getMaxDuration()).isEqualTo(7_500.0);
+        assertThat(completeComputeMetrics.getAverageDuration()).isEqualTo(6_750.0);
+    }
+    // endregion
+
+    // region getCompleteComputeDurations
+    @Test
+    void shouldGetCompleteComputeDurations() {
+        when(appComputeDurationsService.getChainTaskIds()).thenReturn(List.of(CHAIN_TASK_ID_1, CHAIN_TASK_ID_2));
+
+        // First task has completed
+        System.out.println("Task 1 :" + CHAIN_TASK_ID_1);
```

Comment:
minor but `System.out.println` calls should be avoided, even in tests

Reply:
Forgot to delete them after debugging